### ISSUE TITLE
SpreadsheetSelectionModel's .deselect() does the job for an array's first record only and doesn't deselect all the others

### ISSUE THREAD

**Issue by DaniilVeriga** [extnet/Ext.NET#1270]
Discovered on the initial implementation of SpreadsheetSelectionModel (#635).

Sencha issue: [Ext.grid.selection.SpreadsheetModel.deselect() does not support multiple records ](https://www.sencha.com/forum/showthread.php?310467)

**Test Case**
```
<%@ Page Language="C#" %>

<script runat="server">
    protected void Page_Load(object sender, EventArgs e)
    {
        if (!X.IsAjaxRequest)
        {
            Store store = this.GridPanel1.GetStore();
            store.DataSource = new object[]
            {
                new object[] { "id0", "11", "12" },
                new object[] { "id1", "21", "22" },
                new object[] { "id2", "31", "32" },
                new object[] { "id3", "41", "42" },
            };
        }
    }

    protected void DeselectRows(object sender, DirectEventArgs e)
    {
        this.SSM.DeselectRows(new int[] { 0, 1 });
    }
</script>

<!DOCTYPE html>

<html>
<head runat="server">
    <title>Ext.NET v4 Example</title>

    <script src="override.js"></script>
</head>
<body>
    <form runat="server">
        <ext:ResourceManager runat="server" />

        <ext:GridPanel ID="GridPanel1" runat="server">
            <Store>
                <ext:Store runat="server">
                    <Model>
                        <ext:Model runat="server">
                            <Fields>
                                <ext:ModelField Name="id" />
                                <ext:ModelField Name="test1" />
                                <ext:ModelField Name="test2" />
                            </Fields>
                        </ext:Model>
                    </Model>
                </ext:Store>
            </Store>
            <ColumnModel runat="server">
                <Columns>
                    <ext:Column runat="server" Text="Test 1" DataIndex="test1" />
                    <ext:Column runat="server" Text="Test 2" DataIndex="test2" />
                </Columns>
            </ColumnModel>
            <SelectionModel>
                <ext:SpreadsheetSelectionModel
                    ID="SSM"
                    runat="server"
                    ColumnSelect="true"
                    RowSelect="true">
                    <SelectedData>
                        <Rows>
                            <ext:SelectedRow RowIndex="0" />
                            <ext:SelectedRow RowIndex="1" />
                        </Rows>
                    </SelectedData>
                </ext:SpreadsheetSelectionModel>
            </SelectionModel>
        </ext:GridPanel>

        <ext:Button runat="server" Text="DeselectRows" OnDirectClick="DeselectRows" />
    </form>
</body>
</html>
```

Here is the `.deselect()` implementation.
http://docs.sencha.com/extjs/6.0/6.0.1-classic/source/SpreadsheetModel.html#Ext-grid-selection-SpreadsheetModel-method-deselect

```
deselect: function(records, suppressEvent) {
    // API docs are inherited
    var me = this,
        sel = me.selected,
        store = me.view.dataSource,
        len,
        i,
        record,
        changed = false;

    if (sel && sel.isRows) {
        if (!Ext.isArray(records)) {
            records = [records];
        }
        len = records.length;
        for (i = 0; i < len; i++) {
            record = records[i];
            if (typeof record === 'number') {
                record = store.getAt(record);
            }
            changed = changed || sel.remove(record);
        }
    }
    if (changed) {
        me.updateHeaderState();
        if (!suppressEvent) {
            me.fireSelectionChange();
        }
    }
}
```

The problem is here:
`changed = changed || sel.remove(record);`

It is executed only once at the first successful removal. After that `changed` is true and the right part from `||` is not executed.

Fixed during the initial implementation of SpreadsheetSelectionModel. Review after the next ExtJS upgrade - 6.0.2 or later.

**Comment by fabriciomurta** [comment]
Issue still reproducible as of ExtJS 6.0.2 release. Opened an issue on Sencha and linked it on the initial comment.

**Comment by fabriciomurta** [comment]
**Update:** Issue still open after ExtJS 6.5.0 release.

**Comment by fabriciomurta** [comment]
**Update:** Issue still open after ExtJS 6.5.1 release. Issue is acknowledged by Sencha under ID EXTJS-20983.

**Comment by fabriciomurta** [comment]
**Update:** Issue still open after ExtJS 6.5.2 release. Issue is acknowledged by Sencha under ID _EXTJS-20983_.

**Comment by fabriciomurta** [comment]
Ext JS Issue EXTJS-27072 fixed this on 6.6.0. The Sencha issue bound to this unfortunately was left not updated, but tested and can't reproduce the issue with pure Ext JS anymore.

**Comment by fabriciomurta** [comment]
The override was released as of Ext.NET 4.7.0 release.
